src/index.js

```diff
@@ -1,35 +1,24 @@
 var merge = require('broccoli-merge-trees');
-var rollupIntoTree = require('./rollup-tree');
+var rollupIntoTree = require('./rollup-tree').rollupAllTheThings;
+const path = require('path');
+const fs = require('fs-extra');
 
-module.exports = function(modules, indexObj) {
-  var namespacedDependencies = [];
-  var nonNamespacedDependencies = [];
-  for (var i = 0; i < modules.length; i++) {
-    var moduleName = modules[i];
-    var namespaced = true;
-    var name;
-    if (typeof moduleName === 'string') {
-      name = moduleName;
-    } else {
-      name = moduleName.name;
-      namespaced = moduleName.namespaced;
-    }
-
-    var result = {
-      fileName: name.split('/').pop() + '.js',
-      moduleName: name,
-    };
-
-    if (namespaced) {
-      namespacedDependencies.push(result);
-    } else {
-      nonNamespacedDependencies.push(result);
-    }
-  }
+function isPreBuilt(indexObj, preBuildPath) {
+  return !indexObj.isDevelopingAddon && fs.existsSync(preBuildPath);
+}
 
-  if (namespacedDependencies.length > 0) {
+module.exports = function(modules, indexObj) {
+  const classifyDependencies = require('./rollup-tree').classifyDependencies;
+  var dependencies = classifyDependencies(modules);
+  if (dependencies.namespacedDependencies.length > 0) {
     function treeForAddon(root) {
-      return rollupIntoTree.call(this, root, namespacedDependencies, this._super.treeForAddon);
+      if(indexObj.PREBUILT_PATH) {
```

Comment:
rather then duplicating this, lets make `addon.isPrebuilt() { .. }` method, and invoke that method here, and in `treeForVendor`